Unable to create Flex Flow in the Go SDK due to incorrectly generated parameter names

Hey, 

First of all, I just want to say this looks great. Super excited to see the Go SDK enter pilot.

I have been trying out the Go SDK and I ran into an issue when trying to create a Flex Flow. When attempting to create a Flex Flow with the code below

```
package main

import (
    "log"
    "os"

    "github.com/twilio/twilio-go/twilio"
    openapi "github.com/twilio/twilio-go/twilio/rest/flex/v1"
)

func main() {
    accountSid := os.Getenv("TWILIO_ACCOUNT_SID")
    authToken := os.Getenv("TWILIO_AUTH_TOKEN")
    client := twilio.NewClient(accountSid, authToken)

    resp, err := client.FlexV1.CreateFlexFlow(&openapi.CreateFlexFlowParams{
        FriendlyName:    PtrString("Test Flow"),
        ChatServiceSid:  PtrString(os.Getenv("TWILIO_FLEX_CHANNEL_SERVICE_SID")),
        ChannelType:     PtrString("web"),
        IntegrationType: PtrString("external"),
        IntegrationUrl:  PtrString("https://test.com/external"),
    })

    if err != nil {
        log.Panic(err.Error())
    }

    log.Printf("SID: %s", StringValue(resp.Sid))
}

// PtrString is a utility function for obtaining a pointer for a value
func PtrString(value string) *string {
    return &value
}

// StringValue is a utility function for dereferencing a pointer 
func StringValue(ptr *string) string {
    return *ptr
}
```

I get the following error

```
2021/04/15 02:17:47 Status: 400 - ApiError 45004: WebhookUrl is missing (null) More info: https://www.twilio.com/docs/errors/45004
panic: Status: 400 - ApiError 45004: WebhookUrl is missing (null) More info: https://www.twilio.com/docs/errors/45004
```

I was looking through the Go SDK and I noticed the parameter name for Integration URL is incorrect (https://github.com/twilio/twilio-go/blob/main/twilio/rest/flex/v1/api_default.go#L199). The name is set to `IntegrationUrl` whereas in the docs (https://www.twilio.com/docs/flex/developer/messaging/api/flow#create-a-flex-flow-with-studio) and OpenAPI template (https://github.com/twilio/twilio-oai/blob/main/spec/yaml/twilio_flex_v1.yaml#L771) the parameter name is `Integration.Url`. 

This is also applicable to `IntegrationFlowSid`, `IntegrationWorkspaceSid`, `IntegrationWorkflowSid`, `IntegrationChannel`, `IntegrationTimeout`, `IntegrationPriority` and `IntegrationCreationOnMessage` (https://github.com/twilio/twilio-go/blob/main/twilio/rest/flex/v1/api_default.go#L180-L206)

Looking through the generator code in the `api.mustache` template I saw the parameter name is used as the prefix for the Params struct name (https://github.com/twilio/twilio-oai-generator/blob/main/src/main/resources/twilio-go/api.mustache#L34) and as the parameter name which is set as the key in the url Values at https://github.com/twilio/twilio-oai-generator/blob/main/src/main/resources/twilio-go/api.mustache#L81 and https://github.com/twilio/twilio-oai-generator/blob/main/src/main/resources/twilio-go/api.mustache#L84 

The parameter name is modified in `AbstractTwilioGoGenerator` class to remove certain characters (https://github.com/twilio/twilio-oai-generator/blob/main/src/main/java/com/twilio/oai/AbstractTwilioGoGenerator.java#L59) including full stops/ periods. 

The parameter name in the url Values should be the original name, specified in the OpenAPI template and not the modified value. I can't remember which other API's use full stops/ periods in their parameter names but I don't believe it's limited to just Flex Flows. I could be wrong though.

I'm not sure if the problem can be fixed in this repo or would require any upstream changes but I thought this would be the most sensible place to raise it. I’m not sure on the best way to address this issue but I am happy to make any necessary changes.

I didn't see a bug template in the list and I wasn't sure if this was a support case or not, so apologies if this is the wrong place to raise this issue or if it is in the wrong format.

Thanks
Rob

--- a/PULL_REQUEST_TEMPLATE.md
+++ b/PULL_REQUEST_TEMPLATE.md
@@ -20,6 +20,12 @@ A short description of what this PR does.
 
 ### Checklist
 - [x] I acknowledge that all my contributions will be made under the project's license
+- [ ] Run `make test-docker`
+- [ ] Verify affected language:
+    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
+    - [ ] Run `make test` in `twilio-go`
+    - [ ] Create a pull request in `twilio-go`
+    - [ ] Provide a link below to the pull request
 - [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
 - [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
 - [ ] I have titled the PR appropriately

--- a/examples/go/oai.yaml
+++ b/examples/go/oai.yaml
@@ -664,7 +664,7 @@
           - description: 'The `date_test` value, specified as `YYYY-MM-DD`, of the
             resources to read.'
             in: query
-            name: DateTest
+            name: Date.Test
             schema:
               format: date
               type: string

--- a/examples/go/test/api_test.go
+++ b/examples/go/test/api_test.go
@@ -112,7 +112,7 @@ func TestDateInPath(t *testing.T) {
 }
 
 func TestCustomHeaders(t *testing.T) {
-	testHeader := "Test Header"
+	testHeader := "true"
 	testUri := "https://validurl.com"
 	params := &openapi.CreateCallRecordingParams{}
 	params.XTwilioWebhookEnabled = &testHeader

--- a/examples/go/test/unit_test.go
+++ b/examples/go/test/unit_test.go
@@ -31,14 +31,14 @@ func TestPathIsCorrect(t *testing.T) {
 }
 
 func TestAddingHeader(t *testing.T) {
-	testHeader := "Custom Header"
+	testHeader := "true"
 	testUri := "https://validurl.com"
 	params := &openapi.CreateCallRecordingParams{}
 	params.XTwilioWebhookEnabled = &testHeader
 	params.RecordingStatusCallback = &testUri
 
 	expectedHeader := make(map[string]interface{})
-	expectedHeader["XTwilioWebhookEnabled"] = "Custom Header"
+	expectedHeader["X-Twilio-Webhook-Enabled"] = "true"
 
 	mockCtrl := gomock.NewController(t)
 	testClient := NewMockBaseClient(mockCtrl)
@@ -71,9 +71,9 @@ func TestQueryParams(t *testing.T) {
 
 	expectedData := url.Values{}
 	expectedData.Set("DateCreated", fmt.Sprint(dateCreated.Format(time.RFC3339)))
-	expectedData.Set("DateTest", fmt.Sprint(dateTest))
-	expectedData.Set("DateCreatedBefore", fmt.Sprint(dateCreatedBefore.Format(time.RFC3339)))
-	expectedData.Set("DateCreatedAfter", fmt.Sprint(dateCreatedAfter.Format(time.RFC3339)))
+	expectedData.Set("Date.Test", fmt.Sprint(dateTest))
+	expectedData.Set("DateCreated<", fmt.Sprint(dateCreatedBefore.Format(time.RFC3339)))
+	expectedData.Set("DateCreated>", fmt.Sprint(dateCreatedAfter.Format(time.RFC3339)))
 	expectedData.Set("PageSize", fmt.Sprint(pageSize))
 
 	mockCtrl := gomock.NewController(t)

--- a/src/main/resources/twilio-go/api.mustache
+++ b/src/main/resources/twilio-go/api.mustache
@@ -7,6 +7,7 @@ import (
 	"fmt"
 	twilio "github.com/twilio/twilio-go/client"
 	"net/url"
+    "html"
 	{{#imports}}
     {{#-first}}
     {{/-first}}
@@ -78,10 +79,10 @@ func (c *{{{classname}}}Service) {{{nickname}}}({{#allParams}}{{#required}}{{par
 	    return nil, err
 	}
 
-	data.Set("{{paramName}}", string(v))
+	data.Set(html.UnescapeString("{{baseName}}"), string(v))
 {{/isFreeFormObject}}
 {{^isFreeFormObject}}
-    data.Set("{{paramName}}", {{^isArray}}{{^isString}}fmt.Sprint({{/isString}}{{#isDateTime}}({{/isDateTime}}*params.{{paramName}}{{^isString}}{{#isDateTime}}).Format(time.RFC3339){{/isDateTime}}){{/isString}}){{/isArray}} {{#isArray}}strings.Join(*params.{{paramName}}, ",")){{/isArray}}
+    data.Set(html.UnescapeString("{{baseName}}"), {{^isArray}}{{^isString}}fmt.Sprint({{/isString}}{{#isDateTime}}({{/isDateTime}}*params.{{paramName}}{{^isString}}{{#isDateTime}}).Format(time.RFC3339){{/isDateTime}}){{/isString}}){{/isArray}} {{#isArray}}strings.Join(*params.{{paramName}}, ",")){{/isArray}}
 {{/isFreeFormObject}}
 }
 {{/isHeaderParam}}
@@ -91,7 +92,7 @@ func (c *{{{classname}}}Service) {{{nickname}}}({{#allParams}}{{#required}}{{par
 {{#hasHeaderParams}}
 {{#headerParams}}
 	if params != nil && params.{{paramName}} != nil {
-		headers["{{paramName}}"] = *params.{{paramName}}
+		headers["{{baseName}}"] = *params.{{paramName}}
 	}
 {{/headerParams}}
 {{/hasHeaderParams}}

--- a/src/main/resources/twilio-go/api.mustache
+++ b/src/main/resources/twilio-go/api.mustache
@@ -7,7 +7,6 @@ import (
 	"fmt"
 	twilio "github.com/twilio/twilio-go/client"
 	"net/url"
-    "html"
 	{{#imports}}
     {{#-first}}
     {{/-first}}
@@ -79,10 +78,10 @@ func (c *{{{classname}}}Service) {{{nickname}}}({{#allParams}}{{#required}}{{par
 	    return nil, err
 	}
 
-	data.Set(html.UnescapeString("{{baseName}}"), string(v))
+	data.Set("{{{baseName}}}", string(v))
 {{/isFreeFormObject}}
 {{^isFreeFormObject}}
-    data.Set(html.UnescapeString("{{baseName}}"), {{^isArray}}{{^isString}}fmt.Sprint({{/isString}}{{#isDateTime}}({{/isDateTime}}*params.{{paramName}}{{^isString}}{{#isDateTime}}).Format(time.RFC3339){{/isDateTime}}){{/isString}}){{/isArray}} {{#isArray}}strings.Join(*params.{{paramName}}, ",")){{/isArray}}
+    data.Set("{{{baseName}}}", {{^isArray}}{{^isString}}fmt.Sprint({{/isString}}{{#isDateTime}}({{/isDateTime}}*params.{{paramName}}{{^isString}}{{#isDateTime}}).Format(time.RFC3339){{/isDateTime}}){{/isString}}){{/isArray}} {{#isArray}}strings.Join(*params.{{paramName}}, ",")){{/isArray}}
 {{/isFreeFormObject}}
 }
 {{/isHeaderParam}}
@@ -92,7 +91,7 @@ func (c *{{{classname}}}Service) {{{nickname}}}({{#allParams}}{{#required}}{{par
 {{#hasHeaderParams}}
 {{#headerParams}}
 	if params != nil && params.{{paramName}} != nil {
-		headers["{{baseName}}"] = *params.{{paramName}}
+		headers["{{{baseName}}}"] = *params.{{paramName}}
 	}
 {{/headerParams}}
 {{/hasHeaderParams}}

